Source/PersistencyService/PersistencyService.swift

```diff
@@ -1,8 +1,19 @@
 protocol PersistencyService {
+
+    // MARK: - NodeHash
+
     func save(nodeHashes: [NodeHash]) throws
     func nodeHash(forHash hashValue: String) throws -> NodeHash?
```

Comment:
If in the DB there are more than 10.000 records or more it takes time... It could be an improvement to introduce some pagination mechanism

Reply:
will keep it as future improvement.